src/gui/qgsmapcanvas.cpp

```diff
@@ -800,26 +800,62 @@ void QgsMapCanvas::updateMap()
 //the format defaults to "PNG" if not specified
 void QgsMapCanvas::saveAsImage( const QString& theFileName, QPixmap * theQPixmap, const QString& theFormat )
 {
+  QPainter painter;
+  QImage image;
+
   //
   //check if the optional QPaintDevice was supplied
   //
   if ( theQPixmap )
   {
+    image = theQPixmap->toImage();
+    painter.begin( &image );
+
     // render
-    QPainter painter;
-    painter.begin( theQPixmap );
     QgsMapRendererCustomPainterJob job( mSettings, &painter );
     job.start();
     job.waitForFinished();
     emit renderComplete( &painter );
-    painter.end();
-
-    theQPixmap->save( theFileName, theFormat.toLocal8Bit().data() );
   }
   else //use the map view
   {
-    mMap->contentImage().save( theFileName, theFormat.toLocal8Bit().data() );
+    image = mMap->contentImage().copy();
+    painter.begin( &image );
+  }
+
+  // draw annotations
+  QList<QGraphicsItem*> itemList = items();
+  if ( itemList.size() < 1 )
+  {
+    return;
   }
+  QGraphicsItem* item = 0;
```

Comment:
Use `nullptr` (or skip this entirely, the initial assignment will never be used)